optaweb-vehicle-routing-backend/src/main/java/org/optaweb/vehiclerouting/plugin/planner/RouteOptimizerImpl.java

```diff
@@ -199,28 +186,30 @@ public void addLocation(org.optaweb.vehiclerouting.domain.Location coreLocation,
     public void removeLocation(org.optaweb.vehiclerouting.domain.Location coreLocation) {
         Location location = coreToPlanner(coreLocation);
         if (!solver.isSolving()) {
+            if (tsp.getLocationList().size() != 1) {
+                throw new IllegalStateException("Impossible number of locations (" + tsp.getLocationList().size()
+                        + ") when solver is not solving.\n" + tsp.getLocationList());
+            }
             tsp.getLocationList().remove(0);
             tsp.setDomicile(null);
-            updateScore(tsp);
-            sendRoute(tsp);
+            publishRoute(tsp);
         } else {
+            if (tsp.getDomicile().getLocation().getId().equals(location.getId())) {
+                throw new UnsupportedOperationException("You can only remove domicile if it's the only location on map.");
```

Comment:
True, it doesn't exactly the match the condition. But I know that `solver.isSolving() => locations.size > 1` because whenever locations are reduced to 1 in `removeLocation()` method, solver is stopped by calling `terminateEarly()`.